src/App.tsx

```diff
@@ -5,7 +5,19 @@ import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
 
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+
+  function addToTheList(movie: Movie) {
+    setMovies(prev => {
+      const alreadyAdded = prev.some(mov => mov.imdbId === movie.imdbId);
+
+      if (alreadyAdded) {
+        return prev;
+      }
+
+      return [...prev, movie];
+    });
```

Comment:
```suggestion
    const hasMovie = prev.some(mov => mov.imdbId === movie.imdbId);

    if (hasMovie) {
        return;
    }

    setMovies(prev =>[...prev, movie]);
```
You can do the check before setting a state